ta/pkcs11/src/serializer.h

```diff
@@ -22,19 +22,94 @@ struct serialargs {
 	size_t size;
 };
 
+/*
+ * serialargs_init() - Initilize with a new input buffer
+ * @args:	serializing state
+ * @in:		input buffer
+ * @size:	size of the input buffer
+ */
 void serialargs_init(struct serialargs *args, void *in, size_t size);
 
+/*
+ * serialargs_get() - copy out a chunk of data and advance
+ * @args:	serializing state
+ * @out:	output buffer
+ * @sz:		number of bytes to copy to output buffer
+ *
+ * Returns PKCS11_CKR_OK on success or PKCS11_CKR_ARGUMENTS_BAD on failure.
+ */
 enum pkcs11_rc serialargs_get(struct serialargs *args, void *out, size_t sz);
 
+/*
+ * serialargs_get_u32() - copy out a uint32_t and advance
+ * @args:	serializing state
+ * @out:	output buffer
+ * @sz:		number of bytes to copy to output buffer
+ *
+ * Returns PKCS11_CKR_OK on success or PKCS11_CKR_ARGUMENTS_BAD on failure.
+ */
+static inline enum pkcs11_rc serialargs_get_u32(struct serialargs *args,
+						uint32_t *out)
+{
+	return serialargs_get(args, out, sizeof(*out));
+}
+
+/*
+ * serialargs_get_ptr() - get a pointer to a chunk of data and advance
+ * @args:	serializing state
+ * @out:	Pointer to the data retrieved in *@out
```

Comment:
s/in *@out/from @args/

Reply:
Yes it's messy, I don't know how to describe it accurately. I'll leave it as it is until we're able to sort it out.

Reply:
ok